util/util.go

```diff
@@ -221,15 +234,47 @@ func MakeDeploy(
 
 	deploy = &ipc.DeployItem{
 		Address:           fromAddress,
-		Session:           &ipc.DeployPayload{Payload: &ipc.DeployPayload_DeployCode{DeployCode: &ipc.DeployCode{Code: sessionCode, Args: sessionArgs}}},
-		Payment:           &ipc.DeployPayload{Payload: &ipc.DeployPayload_DeployCode{DeployCode: &ipc.DeployCode{Code: paymentCode, Args: paymentArgs}}},
+		Session:           MakeDeployPayload(sessionName, sessionData, sessionArgs),
+		Payment:           MakeDeployPayload(paymentName, paymentData, paymentArgs),
 		GasPrice:          gasPrice,
 		AuthorizationKeys: [][]byte{fromAddress},
 		DeployHash:        headerHash}
 
 	return deploy
 }
 
+func MakeDeployCode(name string, data []byte, args []byte) *consensus.Deploy_Code {
+	deployCode := &consensus.Deploy_Code{AbiArgs: args}
+	switch name {
+	case "wasm":
+		deployCode.Contract = &consensus.Deploy_Code_Wasm{Wasm: data}
+	case "uref":
+		deployCode.Contract = &consensus.Deploy_Code_Uref{Uref: data}
+	case "hash":
+		deployCode.Contract = &consensus.Deploy_Code_Hash{Hash: data}
+	default:
```

Comment:
Does "name" not have to be treated as an case? and "default " clause would handle an error case.

Reply:
https://github.com/hdac-io/casperlabs-ee-grpc-go-util/pull/7/commits/900e086df8a501abf94e88e99bd102297c4a4b94